tests/unit/containers/my-cooperations/add-review-modal/AddReviewModal.spec.jsx

```diff
@@ -0,0 +1,136 @@
+import { describe, expect, vi } from 'vitest'
+import { render, screen } from '@testing-library/react'
+
+import { UserRoleEnum } from '~/types/user/user.index'
+import userEvent from '@testing-library/user-event'
+
+import AddReviewModal from '~/containers/my-cooperations/add-review-modal/AddReviewModal'
+
+const callbackMocks = {
+  dispatch: vi.fn(),
+  closeModal: vi.fn()
+}
+
+const inputProps = {
+  data: {
+    targetUserId: 'userID',
+    targetUserRole: UserRoleEnum.Tutor,
+    offer: 'offerText'
+  }
+}
+
+vi.mock('react-i18next', () => ({
+  useTranslation: () => ({
+    t: (key) => key
+  })
+}))
+
+vi.mock('~/hooks/use-redux', () => ({
+  useAppSelector: () => ({
+    userRole: UserRoleEnum.Tutor
+  }),
+  useAppDispatch: () => callbackMocks.dispatch
+}))
+
+vi.mock('~/context/modal-context', () => ({
+  useModalContext: () => ({
+    closeModal: callbackMocks.closeModal
+  })
+}))
+
+vi.mock('~/hooks/use-mutation', () => ({
+  default: ({ onSuccess, onError }) => ({
+    mutate: (ReviewData) => {
+      if (ReviewData.comment != 'error-trigger') {
+        onSuccess()
+      } else {
+        onError()
+      }
```

Comment:
Why do we need a reverse condition here?
```suggestion
      if (ReviewData.comment === 'error-trigger') {
        onError()
      } else {
        onSuccess()
      }
```